test-e2e/utils.js

```diff
@@ -929,3 +922,150 @@ export async function seedProjectBlobs(project, t, { photoCount, audioCount }) {
+    )
+    assert(icon, `icon ${iconRef.docId} exists for preset ${presetName}`)
+    assert.deepEqual(
+      await project.$icons[kGetIconBlob](iconRef.docId, {
+        mimeType: 'image/svg+xml',
+        size: 'medium',
+      }),
+      Buffer.from(iconsFromConfig.get(icon.name) || '', 'utf-8'),
+      `icon matches config`
+    )
+  }
+
+  // Check all the translations are present
+  for await (const { lang, translations } of reader.translations()) {
+    if (
+      !Object.keys(translations?.field || {}).length &&
+      !Object.keys(translations?.category || {}).length
+    ) {
+      continue
+    }
+    const translatedDocCounts = {
+      category: 0,
+      field: 0,
+    }
+    const translatedPresets = await project.preset.getMany({ lang })
+    const translatedFields = await project.field.getMany({ lang })
+    for (const [docType, translatedDocs] of Object.entries(translations)) {
+      for (const [docId, translatedDoc] of Object.entries(translatedDocs)) {
+        let projectDoc
+        if (docType === 'category') {
+          const category = expectedPresets.get(docId)
+          if (!category) {
+            continue
+          }
+          projectDoc = translatedPresets.find(
+            (p) => p.name === translatedDoc.name
+          )
+        } else if (docType === 'field') {
+          const field = fieldsFromConfig.get(docId)
+          if (!field) continue
+          projectDoc = translatedFields.find((f) => f.tagKey === field.tagKey)
+        } else {
+          continue
+        }
+        if (!projectDoc) continue
+        translatedDocCounts[docType]++
+        for (const [propertyRef, message] of Object.entries(translatedDoc)) {
+          if (propertyRef === 'terms') continue // TODO: support preset.terms
+          assert.equal(
+            getProperty(projectDoc, propertyRef),
+            message,
+            `translated ${docType} ${docId} property ${propertyRef} matches`
+          )
+        }
+      }
+    }
+    // The category archive can include translations which are not imported,
+    // because our import code ignores translations that refer to docs or
+    // properties that do not exist, so we can't match the count here. Instead
+    // we check that at least one translation was checked, as a check that this
+    // test is actually testing something.
+    assert(
+      translatedDocCounts.category > 0,
+      `some ${lang} category translations`
+    )
+    assert(translatedDocCounts.field > 0, `some ${lang} field translations`)
+  }
+}
+
+/**
+ * Create a temporary categories archive from a folder, cleanup after test
+ *
+ * @param {import('node:test').TestContext} t
+ * @param {string} folder
+ * @returns {Promise<string>}
+ */
+export async function categoriesArchiveFromFolder(t, folder) {
+  const archivePath = temporaryFile()
+  t.after(() => fs.unlinkSync(archivePath))
+  await pipeline(
+    execa`npx comapeocat build ${folder}`.readable(),
```

Comment:
Is there no programmatic API we could import that does this? IMO we should make one available.